Formula/llvm.rb

```diff
@@ -444,6 +444,21 @@ def caveats
   end
 
   test do
+    # Use brewed glibc if installed.
+    glibc_args = if OS.linux?
+      glibc = Formula["glibc"]
+      glibc_installed = glibc.any_version_installed?
+      if glibc_installed
+        [
+          "-nostdlib",
+          "-Wl,-dynamic-linker=#{HOMEBREW_PREFIX}/lib/ld.so",
+          "#{Formula["gcc"].opt_lib}/gcc/current/gcc/x86_64-pc-linux-gnu/12/Scrt1.o",
+          "-L#{glibc.opt_lib}",
+          "-lc",
+        ]
+      end
+    end
```

Comment:
Does this mean one has to pass these flags in order to use `llvm` on `ubuntu-20.04`?

Reply:
Based on the tests which pass or don't pass, it appears that these are needed if `libstdc++` or `libc++` are dynamically linked but not statically linked.  Additional flags are needed when using `libstdc++` to make sure it uses the brewed `libstdc++` from `gcc` instead of the host one.

I just found this information on configuration files that look very similar to GCC specs: https://clang.llvm.org/docs/UsersManual.html#configuration-files.  This seems to be new as of LLVM 12 and should allow us to add flags like this to the default configuration file just as we do on GCC.  This may even be useful on macOS to some extent.

Reply:
Yes, I discusses using these at #111193 too, but I think we want some upstream changes (https://reviews.llvm.org/D134337) to land to be able to use it properly. `llvm` will also need to be rebuilt with configuration for `CLANG_CONFIG_FILE_SYSTEM_DIR` or else we will need to put the `.cfg` files next to the `clang` binary.

It sounds like `llvm` is just ignoring Homebrew `glibc` -- does it have the right `RPATH`?

Reply:
> Does this mean one has to pass these flags in order to use `llvm` on `ubuntu-20.04`?

I am also concerned about this. I'd love to see a fix in the formula for LLVM or GLIBC rather than the test as the latter implies that all users will have to make that change too.

Reply:
Yes, it'd be better to fix it in `llvm` / `clang`. I hadn't heard of LLVM config files. That's good news!